### DISCUSSION TITLE
ALLUSERS parameter is ignored by Wix installer

### DISCUSSION THREAD

**Discussion by liangming2003** [wixtoolset/issues#8090] [Questions]
I have very simple Wix installer, it extracts a installer and a script(install.cmd) to c:\Windows\temp and run the script.

If run the Wix installer, the INSTALL.log is like this:

`Property(S): WixPerUserFolder = C:\windows\system32\config\systemprofile\AppData\Local\Apps\Remote Desktop`

If run the install.cmd in command line, the INSTALL.log is like this:

`Property(S): WixPerUserFolder = C:\Users\Admin\AppData\Local\Apps\Remote Desktop`

It seems that ALLUSERS parameter is ignored by Wix installer. Any solution for this problem?

AppInstaller.wxs

<?xml version='1.0'?>
<Wix xmlns='http://schemas.microsoft.com/wix/2006/wi'>
   <Product Id='*' UpgradeCode="3508345-7155-452D-879E-7AC8592FC9B6" 
        Name='AppInstaller' Language='1033' Version='1.0.0.0' 
        Manufacturer='ABC'> 
    
    <Package Description='AppInstaller script' InstallerVersion='200' InstallScope="perMachine" Compressed='yes' />
    <Property Id="AllUSERS" Value="1"/>

    <Media Id='1' Cabinet='setup.cab' EmbedCab='yes' />

    
    <Directory Id='TARGETDIR' Name='SourceDir'>
        <Directory Id="WindowsFolder">
            <Directory Id="INSTALLLOCATION" Name="Temp">
                <Component Id='AppInstallerComponent' DiskId='1' Guid=''>
                    <File Id="File0" Name="RemoteDesktop_1.2.5255.0_x64.msi" Source="Files\RemoteDesktop_1.2.5255.0_x64.msi" />
                </Component>
            </Directory>
        </Directory>

        <Directory Id="TempFolder">
            <Directory Id="tmpdir" Name="~_tmpdir">
                <Component Id='AppInstallScript' DiskId='1' Guid=''>
                    <File Id="File100" Name="install.cmd" Source="Files\install.cmd" />
                </Component>
            </Directory>  
        </Directory>  

        
    </Directory>

    <Feature Id='InstallFeature' Title='Install Feature' Level='1'>
        <ComponentRef Id='AppInstallerComponent' />
        <ComponentRef Id='AppInstallerScript' />
    </Feature>

    
    <CustomAction Id="RunWrapExe" Return="asyncNoWait" Execute="deferred" 
                  FileKey="File100" ExeCommand=""
                  HideTarget="no" Impersonate="no" />

    <InstallExecuteSequence>
        <Custom Action="RunWrapExe" 
                After="InstallFiles">NOT REMOVE~="ALL"</Custom>
        <PublishComponents Suppress="yes" />
        <PublishFeatures Suppress="yes" />
        <PublishProduct Suppress="yes" />
        <RegisterProduct Suppress="yes" />
        <RegisterUser Suppress="yes" />
    </InstallExecuteSequence>

   </Product>
</Wix>

install.cmd

```
REM The following is required in all INSTALL.CMD files
Call c:\system.sav\util\SetVariables.cmd
Set block=%~dp0
Set TMP=C:\Windows\Temp
Set TEMP=%TMP%
 
REM Install Remote Desktop & capture logs
start /wait msiexec.exe **ALLUSERS=1** /i C:\Windows\Temp\RemoteDesktop_1.2.5255.0_x64.msi /l* C:\Windows\Temp\INSTALL.log /quiet /norestart

REM if %ERRORLEVEL% equ 0 del /F /Q C:\Windows\Temp\RemoteDesktop_1.2.5255.0_x64.msi
```

**Comment by liangming2003** [comment]
logs comparison:
![ALLUSERS](https://github.com/wixtoolset/issues/assets/13327028/8b8e3ec4-3339-4a11-b69d-356023bbf374)

**Comment by chrpai** [comment]
You claim to be using an MSI to lay down another MSI and install it from the InstallExecute sequence in deferred no impersonation context. That simply isn't possible due to the _MSIMutex limitation of Windows Installer.

You should look at Burn, WiX's bootstrapper/bundler. It's an EXE and can cache and install MSIs like this scenario.
